src/components/Rocket.js

```diff
@@ -0,0 +1,91 @@
+import '../styles/Rocket.css';
+import PropTypes from 'prop-types';
+import { useEffect, useState, useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { buttonReducer } from '../redux/rockets/RocketSlice';
+import getRockets from '../redux/rockets/RocketAPI';
+
+const Rocket = ({
+  id, imgURL, title, description,
+}) => {
+  const [buttonText, setButtonText] = useState('Reserve Rocket');
+  const { reserveState } = useSelector((store) => store.rocket);
+  const buttonRef = useRef(null);
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    reserveState.forEach((item) => {
+      if (item.id === id) {
+        setButtonText('Cancel Reservation');
+        buttonRef.current.classList.add('buttonCancel');
+      }
+    });
+  }, [id, reserveState]); // Added 'id' and 'reserveState' to dependency array
+
+  function setButton(e) {
+    if (buttonText === 'Reserve Rocket') {
+      setButtonText('Cancel Reservation');
+      dispatch(buttonReducer({ id, buttonText, title }));
+      e.target.classList.add('buttonCancel');
```

Comment:
- [ ] [OPTIONAL] The reserved rockets should show a "Reserved" badge when the `Reserve Rocket` button is clicked on the left of the description as seen in the design. Kindly ensure to implement this ☺️.

Template design 😎 
![image](https://user-images.githubusercontent.com/99394374/232031216-2d728a21-1904-44c4-a804-0214ade01985.png)

Your design 😎 
![image](https://user-images.githubusercontent.com/99394374/232031300-5c2ee289-ac70-4b69-af8b-2438246b02e3.png)